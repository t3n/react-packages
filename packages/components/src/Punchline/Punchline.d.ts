declare const Punchline: import("styled-components").StyledComponent<"h1", any, import("../Heading").HeadingProps, never>;
export default Punchline;
