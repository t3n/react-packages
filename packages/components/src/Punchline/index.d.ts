import Punchline from './Punchline';
export * from './Punchline';
export default Punchline;
