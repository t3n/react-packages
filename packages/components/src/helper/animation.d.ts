import { ThemeProps } from '@t3n/theme';
export declare const backgroundAnimation: import("styled-components").FlattenInterpolation<import("styled-components").ThemedStyledProps<ThemeProps, any>>;
