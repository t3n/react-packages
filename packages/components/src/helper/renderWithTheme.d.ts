import 'jest-styled-components';
import React from 'react';
import { RenderOptions } from '@testing-library/react';
import { Theme } from '@t3n/theme';
interface OptionalTheme {
    theme?: Theme;
}
export declare const renderWithTheme: (ui: React.ReactElement, { theme, ...options }: RenderOptions & OptionalTheme) => import("@testing-library/react").RenderResult<typeof import("@testing-library/dom/types/queries"), HTMLElement>;
export {};
