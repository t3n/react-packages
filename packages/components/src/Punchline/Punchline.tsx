import styled from 'styled-components';
import { color, space, typography } from 'styled-system';

import { Heading } from '../Heading';

export const Punchline = styled(Heading)`
  ${({ theme }) =>
    typography({
      theme,
      fontSize: [8, '3rem', '3.5rem', '4rem'],
      textAlign: 'left',
      lineHeight: ['1.5em', '1.5em', '1.6em', '1.6em'],
    })};
  ${({ theme }) => space({ theme, pb: [0, 0, 1, 1] })};
  ${({ theme }) => color({ theme, color: 'text.inverse', bg: 'shades.black' })};

  box-shadow: 0.625rem 0 0 #000, -0.625rem 0 0 #000;
  display: inline;
`;
