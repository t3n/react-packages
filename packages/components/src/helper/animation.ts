import { css, keyframes } from 'styled-components';

import { ThemeProps } from '@t3n/theme';

const backgroundAnimationKeyframes = ({ theme }: ThemeProps) => keyframes`
  0% {
    background-color: ${theme.colors.shades.grey204}
  }

  50% {
    background-color: ${theme.colors.shades.grey244}
  }

  100% {
    background-color: ${theme.colors.shades.grey204}
  }
`;

// eslint-disable-next-line import/prefer-default-export
export const backgroundAnimation = css`
  animation: ${backgroundAnimationKeyframes} 2s linear infinite;
`;
