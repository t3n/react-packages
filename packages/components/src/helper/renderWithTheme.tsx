/* eslint-disable import/no-extraneous-dependencies */
import 'jest-styled-components';

import React from 'react';
import { render, RenderOptions } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';

import { Theme, theme as originalTheme } from '@t3n/theme';

interface OptionalTheme {
  theme?: Theme;
}

// eslint-disable-next-line import/prefer-default-export
export const renderWithTheme = (
  ui: React.ReactElement,
  { theme, ...options }: RenderOptions & OptionalTheme
) => {
  // merge over default theme

  const mergedTheme = { ...originalTheme, ...theme };

  const Wrapper: React.FC = ({ children }) => (
    <ThemeProvider theme={mergedTheme}>{children}</ThemeProvider>
  );

  return render(ui, { wrapper: Wrapper, ...options });
};
