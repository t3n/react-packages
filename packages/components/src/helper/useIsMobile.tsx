import { Theme } from '@t3n/theme';
import { useCallback, useContext, useEffect, useState } from 'react';
import { ThemeContext } from 'styled-components';

const useIsMobile = (): boolean => {
  const [isMobile, setIsMobile] = useState(false);

  const theme: Theme = useContext(ThemeContext);
  const fontSize = window
    .getComputedStyle(window.document.body)
    .getPropertyValue('font-size');
  const mobileBreakpoint =
    parseInt(theme.breakpoints[1], 10) * parseInt(fontSize, 10);

  const testIsMobile = useCallback(() => {
    if (window.matchMedia(`(max-width: ${mobileBreakpoint}px)`).matches) {
      setIsMobile(true);
    } else {
      setIsMobile(false);
    }
  }, [mobileBreakpoint]);

  useEffect(() => {
    testIsMobile();
    window.addEventListener('resize', testIsMobile);

    return () => {
      window.removeEventListener('resize', testIsMobile);
    };
  }, [testIsMobile]);

  return isMobile;
};

export default useIsMobile;
