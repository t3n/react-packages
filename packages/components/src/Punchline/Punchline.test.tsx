import React from 'react';

import { renderWithTheme } from '../helper/renderWithTheme';
import { Punchline } from './Punchline';

test('Punchline machtes snapshot', () => {
  const { container } = renderWithTheme(
    <>
      <Punchline>Test</Punchline>
    </>,
    {}
  );

  expect(container.firstChild).toMatchSnapshot();
});
