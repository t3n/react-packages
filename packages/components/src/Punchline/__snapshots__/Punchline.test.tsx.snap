// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Punchline machtes snapshot 1`] = `
.c0 {
  font-family: "adelle-sans",sans-serif;
  font-size: 1.5rem;
  line-height: 1.333em;
  font-size: 2.5rem;
  text-align: left;
  line-height: 1.5em;
  padding-bottom: 0;
  color: #fff;
  background-color: #000;
  box-shadow: 0.625rem 0 0 #000,-0.625rem 0 0 #000;
  display: inline;
}

@media screen and (min-width:30rem) {
  .c0 {
    font-size: 1.75rem;
  }
}

@media screen and (min-width:48rem) {
  .c0 {
    font-size: 2rem;
  }
}

@media screen and (min-width:61.25rem) {
  .c0 {
    font-size: 2.25rem;
  }
}

@media screen and (min-width:90rem) {
  .c0 {
    font-size: 2.5rem;
  }
}

@media screen and (min-width:30rem) {
  .c0 {
    line-height: 1.333em;
  }
}

@media screen and (min-width:48rem) {
  .c0 {
    line-height: 1.333em;
  }
}

@media screen and (min-width:61.25rem) {
  .c0 {
    line-height: 1.25em;
  }
}

@media screen and (min-width:90rem) {
  .c0 {
    line-height: 1.25em;
  }
}

@media screen and (min-width:30rem) {
  .c0 {
    font-size: 3rem;
    line-height: 1.5em;
  }
}

@media screen and (min-width:48rem) {
  .c0 {
    font-size: 3.5rem;
    line-height: 1.6em;
  }
}

@media screen and (min-width:61.25rem) {
  .c0 {
    font-size: 4rem;
    line-height: 1.6em;
  }
}

@media screen and (min-width:30rem) {
  .c0 {
    padding-bottom: 0;
  }
}

@media screen and (min-width:48rem) {
  .c0 {
    padding-bottom: 4px;
  }
}

@media screen and (min-width:61.25rem) {
  .c0 {
    padding-bottom: 4px;
  }
}

<h1
  class="c0"
>
  Test
</h1>
`;
